sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -179,6 +179,33 @@ func (c *connection) String() string {
 	return fmt.Sprintf("%s: %s %s %s", c.containerID, c.local, arrow, c.remote)
 }
 
+// IsExternal returns true when IPv4 does not belong to the private IP addresses; false otherwise.
+// Error is returned when IP address is malformed
+func (c *connection) IsExternal() (bool, error) {
+	addr, err := c.getRemoteIPAddress()
+	if err != nil {
+		return true, errors.Wrap(err, "unable to determine if flow is external or internal")
+	}
+	return addr.IsPublic(), nil
+}
+
+// getIPAddress returns the IP address of the connection remote.
+// If that IP is unset, it returns the address of the IP Network to which the remote belongs.
+// If both are unavaliable, an error is returned.
+// This check of both is required, because Collector reports the IP addresses
+// either as IPAndPort.Address, or IPAndPort.IPNetwork. The former is used in most cases, but sometimes
+// (usually on OCP) the latter is provided. Analyzing only one of those two sources may lead to incorrectly reporting
+// a connection as external on the network graph.
+func (c *connection) getRemoteIPAddress() (net.IPAddress, error) {
+	if c.remote.IPAndPort.IsAddressValid() {
+		return c.remote.IPAndPort.Address, nil
+	}
+	if c.remote.IPAndPort.IPNetwork.IsValid() {
+		return c.remote.IPAndPort.IPNetwork.IP(), nil
```

Comment:
For CIDR blocks, an IPNetwork is going to be returned, with:
- the IP attribute filled with the prefix bits of the CIDR,
- the prefix set to something potentially smaller than the length of the address family.

As long as the remaining bits are set to 0 (which is guaranteed, I think), and the customer does not declare CIDR blocks in private networks (:shrug: ), we should be safe to compare the address with the known private set.

Reply:
> and the customer does not declare CIDR blocks in private networks

In that case, Collector report the same data to Sensor, right?
So the only issue would be that we place that edge in the "Internal Entities" instead of the named CIDR that the customer wished for.
I don't think this affects us, but let me be sure and test exactly that.

Reply:
Since the IP address corresponding to the full prefix (with zeros in the undefined area) is the lowest IP in the subnet range; then the private networks would match for the subnet itself, that's right.

Reply:
I added a case like that to the unit tests: https://github.com/stackrox/stackrox/pull/9718/files#diff-bdc111f86b91adbfb88275e12b1de7bda68012b3a6c13e4c84c4929f4b0fb445R1025

Reply:
> the customer does not declare CIDR blocks in private networks (:shrug: ), we should be safe to compare the address with the known private set.

I tried defining a custom CIDR block as `172.30.0.0/16` and `172.30.0.0/24` but it was never shown on the graph. All connections that were going to that Network were represented as "Internal Entities". Is this the correct behavior @ovalenti ?